temporal-sdk/src/main/java/io/temporal/worker/WorkflowImplementationOptions.java

```diff
@@ -174,6 +178,26 @@ public Builder setDefaultNexusServiceOptions(NexusServiceOptions defaultNexusSer
       return this;
     }
 
+    /**
+     * Enable upserting version search attributes on {@link Workflow#getVersion}. This will cause
+     * the SDK to automatically add the <b>TemporalChangeVersion</b> search attributes to the
+     * workflow when getVersion is called. This search attribute is a keyword list of all the
+     * getVersion calls made in the workflow. The format of each entry is "<Change ID>-<Version>".
+     * This allows for easy discovery of what versions are being used in your namespace.
+     *
+     * <p>Note: This change is backwards compatible, so it is safe to enable or disable this options
+     * with running workflows. However, if this options is enabled, it is not safe to rollback to a
+     * previous version of the SDK that does not support this option.
+     *
+     * <p>The default value is false. This may change in future releases.
```

Comment:
Wonder if we need an issue to track enabling this by default after a release or two